src/main/java/edu/kis/powp/jobs2d/command/CommandTransformation.java

```diff
@@ -0,0 +1,51 @@
+package edu.kis.powp.jobs2d.command;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public abstract class CommandTransformation implements CommandVisitorInterface {
+    private List<DriverCommand> commands;
+
+    public CommandTransformation(){
+        commands = new ArrayList<>();
+    }
+
+    public ICompoundCommand getTransformedCommand() {
+        return new ICompoundCommand() {
+            @Override
+            public Iterator<DriverCommand> iterator() {
+                return commands.iterator();
+            }
+
+            @Override
+            public void execute(Job2dDriver driver) {
+                commands.forEach(c -> c.execute(driver));
```

Comment:
It will be not safe to return a command containing a reference to a command managed by the visitor object. A copy would be much better.